test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -9820,10 +9873,7 @@ TEST_P(EdsTest, EdsServiceNameDefaultsToClusterName) {
 
 class TimeoutTest : public BasicTest {
  protected:
-  void SetUp() override {
-    xds_resource_does_not_exist_timeout_ms_ = 500;
-    BasicTest::SetUp();
-  }
+  TimeoutTest() { xds_resource_does_not_exist_timeout_ms_ = 500; }
```

Comment:
This isn't going to work.  The test requires that `xds_resource_does_not_exist_timeout_ms_` be set before the parent class' `SetUp()` method is called, since this value is used to create the client stub.

I see a few options here:
- Go back to using `SetUp()` and `TearDown()` instead of the ctor/dtor.
- Add a call to `ResetStub()` here after setting `xds_resource_does_not_exist_timeout_ms_`.  This is a little wasteful, since it will initialize the client channel twice.
- Add an optional parameter to the `XdsEnd2endTest` ctor to set this value, and pass it here via the initialization list.  This would have to be plumbed through `BasicTest`, which is a little invasive.

I think that for now, going back to using `SetUp()` and `TearDown()` is probably the best option.  We can probably improve this later, because in order to split this up into multiple files, we're going to have to provide a cleaner API for subclasses that does not involve reaching directly into the parent class's internals.

Reply:
I simply changed `TimeoutTest` to inherit from `XdsEnd2endTest` instead and added `xds_resource_does_not_exist_timeout_ms` as an optional parameter to the `XdsEnd2endTest` ctor which seems like the right thing to do here even considering future changes.